cardano-node-chairman/test/Test/Cardano/Node/Chairman/Shelley.hs

```diff
@@ -0,0 +1,413 @@
+
+module Test.Cardano.Node.Chairman.Shelley
+  ( tests
+  ) where
+
+import           Chairman.IO.Network.Sprocket (Sprocket (..))
+import           Control.Concurrent.Async
+import           Control.Monad
+import           Control.Monad.IO.Class (liftIO)
+import           Data.Aeson
+import           Data.Bool
+import           Data.Char
+import           Data.Either
+import           Data.Function
+import           Data.Functor
+import           Data.Int
+import           Data.Map (Map)
+import           Data.Maybe
+import           Data.Ord
+import           Data.Semigroup
+import           Data.String (String)
+import           GHC.Float
+import           GHC.Num
+import           Hedgehog (Property, discover, (===))
+import           Lens.Micro
+import           Lens.Micro.Aeson
+import           System.Exit (ExitCode (..))
+import           System.IO (IO)
+import           Text.Read
+import           Text.Show
+
+import qualified Chairman.Base as H
+import qualified Chairman.IO.File as IO
+import qualified Chairman.IO.Network.Socket as IO
+import qualified Chairman.IO.Network.Sprocket as IO
+import qualified Chairman.Process as H
+import qualified Chairman.String as S
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.List as L
+import qualified Data.Map.Lazy as M
+import qualified Data.Time.Clock as DTC
+import qualified Hedgehog as H
+import qualified System.Directory as IO
+import qualified System.Environment as IO
+import qualified System.FilePath.Posix as FP
+import qualified System.IO as IO
+import qualified System.Process as IO
+
+{- HLINT ignore "Redundant <&>" -}
+
+rewriteConfiguration :: String -> String
+rewriteConfiguration "Protocol: RealPBFT" = "Protocol: TPraos"
+rewriteConfiguration "minSeverity: Info" = "minSeverity: Debug"
+rewriteConfiguration "TraceBlockchainTime: False" = "TraceBlockchainTime: True"
+rewriteConfiguration s = s
+
+rewriteGenesisSpec :: Int -> Value -> Either String Value
+rewriteGenesisSpec supply v = Right $ v
+  & key "slotLength" .~ toJSON @Double 0.2
+  & key "activeSlotsCoeff" .~ toJSON @Double 0.1
+  & key "securityParam" .~ toJSON @Int 10
+  & key "epochLength" .~ toJSON @Int 1500
+  & key "maxLovelaceSupply" .~ toJSON supply
+  & key "protocolParams" . key "decentralisationParam" .~ toJSON @Double 0.7
+
+prop_spawnShelleyCluster :: Property
+prop_spawnShelleyCluster = H.propertyOnce . H.workspace "chairman" $ \tempAbsPath -> do
+  tempBaseAbsPath <- H.noteShow $ FP.takeDirectory tempAbsPath
+  tempRelPath <- H.noteShow $ FP.makeRelative tempBaseAbsPath tempAbsPath
+  base <- H.noteShowM H.getProjectBase
+  baseConfig <- H.noteShow $ base <> "/configuration/chairman/defaults/simpleview"
+  currentTime <- H.noteShowIO DTC.getCurrentTime
+  startTime <- H.noteShow $ DTC.addUTCTime 10 currentTime -- 10 seconds into the future
+  socketDir <- H.noteShow $ tempRelPath <> "/socket"
+  env <- H.evalIO IO.getEnvironment
+
+  let bftNodes = ["node-bft1", "node-bft2"] :: [String]
```

Comment:
This is was copied straight out of on `mkfiles.sh`.  Maybe that needs updating too?

Reply:
`mkfiles.sh` starts a cluster in Byron which can then transition to Shelley, so the naming there makes sense. The Byron era is confusing because OBFT was needed for the hardfork to Shelley.

https://github.com/input-output-hk/cardano-node/blob/master/scripts/byron-shelley/mkfiles.sh#L6

Reply:
There's another `mkfiles.sh` script it turns out under `shelley-from-scratch`:

https://github.com/input-output-hk/cardano-node/blob/42dba1b1a3d8a0dccaeab473372c0f8929eff4df/scripts/shelley-from-scratch/mkfiles.sh#L8

That's the one this PR is based on.

Reply:
Should I rename `bft` to `praos`?